### ISSUE TITLE
npm-cache checksum downloaded dependencies

### ISSUE THREAD

**Issue by jamesongithub** [swarajban/npm-cache#19]
we're seeing this error every few days in our ci

```
[npm-cache] [INFO] [npm] extracting dependencies from /path/to/caches/npm/2.3.0/fe56bf256353b99775417bb4904da265.tar.gz 

gzip: stdin: decompression OK, trailing garbage ignored 
tar: Child returned status 2 
tar: Error is not recoverable: exiting now 
[npm-cache] [ERROR] [npm] error untar-ing /path/to/caches/npm/2.3.0/fe56bf256353b99775417bb4904da265.tar.gz 
[npm-cache] [ERROR] error installing dependencies 

Fatal error: local() encountered an error (return code 1) while executing 'npm-cache install --cacheDirectory /path/to/caches/' 
```

it seems like file is corrupted during download?
npm-cache should checksum downloaded dependencies?

**Comment by swarajban** [comment]
Wow that's strange, can you give me some more info about your OS & package/bower/composer.json?

**Comment by jamesongithub** [comment]
Ubuntu 14.04.3 LTS
just use package.json and bower.json
what details do you need?

**Comment by swarajban** [comment]
Can you share your package.json and bower.json so I can try to repro on an
Ubuntu box? Also what npm and node version are you on? I haven't tested
with v4 yet

**Comment by jamesongithub** [comment]
npm 2.12.1
node 0.12.7

**Comment by swarajban** [comment]
Ok, I'll see how far I get w/ that. Will let ya know

**Comment by eugeneiiim** [comment]
Note that we are regularly running npm-cache in multiple concurrent builds. Not sure if that would be problematic.

**Comment by jamesongithub** [comment]
That's a good point. It could be a combination of that plus something else since the rate of occurrence is so high,

**Comment by STRML** [comment]
Seeing this too on our CI. Any clues?

**Comment by swarajban** [comment]
Sorry, haven't had time to look into this yet :(

**Comment by minichate** [comment]
https://github.com/swarajban/npm-cache/pull/57 will fix this issue

**Comment by swarajban** [comment]
@jamesongithub, the latest release, `0.6.2`, should fix this issue thanks to @minichate 's PR #58
